Source/NSStoryboard.m

```diff
@@ -0,0 +1,975 @@
+
+                      // Actions
+                      NSArray *cons = [document nodesForXPath: @"//action" error: NULL];
+                      NSEnumerator *consen = [cons objectEnumerator];
+                      NSXMLElement *celem = nil;
+
+                      while ((celem = [consen nextObject]) != nil)
+                        {
+                          NSXMLNode *targetAttr = [celem attributeForName: @"target"];
+                          NSString *val = [targetAttr stringValue];
+                          if ([val isEqualToString: originalId])
+                            {
+                              [targetAttr setStringValue: @"-1"];
+                            }
+                        }
+
+                      // Outlets
+                      cons = [document nodesForXPath: @"//outlet" error: NULL];
+                      consen = [cons objectEnumerator];
+                      celem = nil;
+                      while ((celem = [consen nextObject]) != nil)
+                        {
+                          NSXMLNode *attr = [celem attributeForName: @"destination"];
+                          NSString *val = [attr stringValue];
+                          if ([val isEqualToString: originalId])
+                            {
+                              [attr setStringValue: @"-1"];
+                            }
+                        }
+                    }
+                }
+
+              // Delete extra firstResponder...
+              /*
+              NSArray *ar = [document nodesForXPath: @"//customObject[@sceneMemberID=\"firstResponder\"]"
+                                              error: NULL];
+              if ([ar count] > 0)
+                {
+                  id node = [ar objectAtIndex: 0];
+                  [node detach];
+                }
+              */
+              
+              // Create document...
+              [_scenesMap setObject: document
+                             forKey: sceneId];
+              
+              // Map controllerId's to scenes...
+              if (controllerId != nil)
+                {
+                  [_controllerMap setObject: sceneId
+                                     forKey: controllerId];
+                }
+              
+              RELEASE(document);
+            }
+        }
+    }
+  else
+    {
+      NSLog(@"No document element in storyboard file");
+    }
+}
+
+- (NSMapTable *) _processSegues: (NSXMLDocument *)xmlIn
+{
+  NSMapTable *mapTable = [NSMapTable strongToWeakObjectsMapTable];
+  NSArray *connectionsArray = [xmlIn nodesForXPath: @"//connections"
+                                             error: NULL];
+  NSArray *array = [xmlIn nodesForXPath: @"//objects"
+                                  error: NULL];
+  NSXMLElement *objects = [array objectAtIndex: 0]; // get the "objects" section
+  NSArray *controllers = [objects nodesForXPath: @"windowController"
```

Comment:
Shouldn't this be moved down where the value gets used for the first time?

Reply:
Done.